api/views/pois.py

```diff
@@ -113,35 +115,36 @@ def update_poi(poi_id):
     data = request.get_json()
 
     # Edit poi columns
-    if 'name' in data:
-        poi.name = data['name']
-    if 'date' in data:
-        poi.date = parse(data['date'])
-    if 'description' in data:
-        poi.description = data['description']
+    poi_columns = ['name', 'date', 'description']
+    edit_poi_columns = [col for col in poi_columns if data.get(col) is not None]
+    for col in edit_poi_columns:
+        setattr(poi, col, data[col])
     
     # Replace all links, media, and story_id if they were given
-    if 'links' in data:
+    if data.get('links') is not None:
         for link in Link.query.filter(Link.poi_id == poi_id):
             db.session.delete(link)
         link_add = [row_constructor(Link, i, poi_id=poi_id) for i in data['links']]
-        db.session.add_all(link_add)
-    if 'media' in data:
+        if len(link_add):
+            db.session.add_all(link_add)

```

Comment:
whats the behavior of `db.session.add_all` if called with an empty list? aka do we really need the `if len(link_add)` check?

Reply:
just tested it and it's not needed since `db.session.add_all` seems to ignore empty lists